### PR TITLE
Clean up warnings.

### PR DESCRIPTION
**Pull request by thealmarty**

### CODE REVIEW

File: library/Core/src/Juvix/Core/Erasure/Algorithm.hs
Comment (thealmarty) [inline library/Core/src/Juvix/Core/Erasure/Algorithm.hs:83]: this is not used, and is the only thing that calls `piTypeToList`. @andy-morris can we delete these lines?

Comment (thealmarty) [inline library/Core/src/Juvix/Core/Erasure/Algorithm.hs:83]: This is not used, and is the only thing that calls `listToPiType`. @andy-morris can we delete this line?

File: library/Core/src/Juvix/Core/Erasure/Algorithm.hs
Comment (thealmarty) [inline library/Core/src/Juvix/Core/Erasure/Algorithm.hs:155]: Not called anywhere, except mentioned above. Can we delete these lines if there are no future plans?

**Review by cwgoes — APPROVED** [review]
ACK modulo the question for @andy-morris

File: library/Core/src/Juvix/Core/Erasure/Algorithm.hs
Comment (thealmarty) [reply library/Core/src/Juvix/Core/Erasure/Algorithm.hs:155]: If there are no future plans, please delete all commented out lines before merging this. Thanks!

**Comment by andy-morris** [comment]
removed the commented parts, as well as some redundant constraints, etc. this is ok to merge unless someone wants to have another quick look at it

**Pull request merged**
